src/generate.ts

```diff
@@ -0,0 +1,26 @@
+export const csvGenerateRow = (
+  row: any,
+  headerKeys: string[],
+  delimiter: string
+) =>
+  headerKeys.map(
+    (fieldName) => typeof row[fieldName] === 'number'
+      ? row[fieldName]
+      : `"${String(row[fieldName]).replace(/"/g, '""')}"`
```

Comment:
numbers are inserted without any processing.  other datatypes are cast to string, have any double-quote characters escaped by adding a second double-quote character, and then the entire string is wrapped in double-quotes similar to how `JSON.stringify()` does.

Reply:
Nice!